[scroll-animations] Make animations/sda-crash.html a wpt

nmoucht opened the pull request:
#### 24ba5bcb3f5457fc4a083a85f956a3f9685dba12
<pre>
[scroll-animations] Make animations/sda-crash.html a wpt
<a href="https://bugs.webkit.org/show_bug.cgi?id=292954">https://bugs.webkit.org/show_bug.cgi?id=292954</a>
<a href="https://rdar.apple.com/151251460">rdar://151251460</a>

Reviewed by Antoine Quint.

Convert newly added test animations/sda-crash.html to be a wpt.

* LayoutTests/TestExpectations:
* LayoutTests/animations/sda-crash-expected.txt: Removed.
* LayoutTests/animations/sda-crash.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/crashtests/scroll-timeline-completion-crash.html: Added.
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/294918@main">https://commits.webkit.org/294918@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0ecbe22c177f23c695a89155328093cb99c66ff3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/103513 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23195 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/13515 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/108689 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54158 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/23547 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/31746 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78657 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/106519 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18256 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/93377 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58991 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/18073 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/11424 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/53514 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/87865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/11483 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111067 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/30660 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/22613 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87651 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31021 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/89577 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87293 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32172 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9886 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25011 "Hash 0ecbe22c for PR 45330 does not build (failure)") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16788 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/30588 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/30388 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/33719 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/31949 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

graouts reviewed (approved):
Nice. Please make sure to export this test.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/scroll-animations/crashtests/scroll-timeline-completion-crash.html:
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<title>This test passes if it does not crash</title>
+<link rel="help" src="https://drafts.csswg.org/scroll-animations-1/#scroll-timeline-shorthand">
+<link rel="help" src="https://drafts.csswg.org/css-conditional-5/#container-queries">
```
I don't see any uses of container queries, can you remove this link?

graouts replied:
Also the specific scroll-animations link is not really relevant. I think you can just link to `https://drafts.csswg.org/scroll-animations-1/`.

nmoucht replied:
Fixed.

nmoucht committed:
[scroll-animations] Make animations/sda-crash.html a wpt
https://bugs.webkit.org/show_bug.cgi?id=292954
rdar://151251460

Reviewed by Antoine Quint.

Convert newly added test animations/sda-crash.html to be a wpt.

* LayoutTests/TestExpectations:
* LayoutTests/animations/sda-crash-expected.txt: Removed.
* LayoutTests/animations/sda-crash.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/crashtests/scroll-timeline-completion-crash.html: Added.
* LayoutTests/platform/mac-wk2/TestExpectations:

Canonical link: https://commits.webkit.org/294918@main
